### PR TITLE
Create partialuser.js

### PR DESCRIPTION
**Pull request by Noxillio**
Using this argument will allow users to specify **part of a username**, along with **all** other types of identifying a user, including full username, full tag, user ID, and mention.

Works the same as Akairo's method of resolving users.

### CODE REVIEW

File: arguments/partialuser.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+module.exports = class extends Argument {
+
+	async run(arg) {
+		const user = await this.resolveUser(arg, this.client.users);
+		if (user) return user;
+		throw `Please specify a valid user.`;
+	}
+
+	resolveUser(text, users, caseSensitive = false, wholeWord = false) {
```
Comment (Tylertron1998) [inline arguments/partialuser.js]: caseSensitive and wholeWord aren't used here, remove them and all code relating to this

File: arguments/partialuser.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const { Argument } = require('klasa');
+
+module.exports = class extends Argument {
+
+	async run(arg) {
+		const user = await this.resolveUser(arg, this.client.users);
+		if (user) return user;
+		throw `Please specify a valid user.`;
```
Comment (Tylertron1998) [inline arguments/partialuser.js]: throw new Error('Please specify a valid user.');

Comment (Noxillio) [reply arguments/partialuser.js]: Why use that when just using "throw" makes it prompt for a valid argument?

Comment (Tylertron1998) [reply arguments/partialuser.js]: Whoops, good point

File: arguments/partialuser.js (hunk 1 above)
Comment (Noxillio) [reply arguments/partialuser.js]: done

**Comment by Noxillio** [comment]
> Just one random though, what does this have that the [`username`](https://github.com/dirigeants/klasa-pieces/blob/master/arguments/username.js) argument doesn't?

You don't have to type the whole username; just part of it will do fine.

**Comment by kyranet** [comment]
Not really, that one matches partial usernames too, it was part of an old version of a username search from my main bot, and served its purpose well.

**Comment by kyranet** [comment]
Hey! What's your status, did the username `argument` work?

**Comment by kyranet** [comment]
Closing due to lack of response. Feel free to re-open!

**Pull request closed without merging**